shipper.py

```diff
@@ -226,49 +227,108 @@ def shipment_post_new():
-                    if os.path.isdir(compendium_files):
-                        file_name = '.'.join((str(data['compendium_id']), 'zip'))
-                        if data['recipient'] == 'zenodo':
-                            data['deposition_id'] = zenodo_create_depot(env_repository_zenodo_host, env_repository_zenodo_token)
-                            data['deposition_url'] = ''.join((env_repository_zenodo_host.replace('api', 'deposit/'), data['deposition_id']))
-                            zenodo_add_zip_to_depot(env_repository_zenodo_host, data['deposition_id'], file_name, compendium_files, env_repository_zenodo_token)
-                            # Add metadata that are in compendium in db:
-                            if 'metadata' in current_compendium:
-                                if 'zenodo' in current_compendium['metadata']:
-                                    md = current_compendium['metadata']['zenodo']
-                                    zenodo_add_metadata(env_repository_zenodo_host, data['deposition_id'], md,
-                                                        env_repository_zenodo_token)
-                        elif data['recipient'] == 'eudat':
-                            data['deposition_id'] = eudat_create_depot(env_repository_eudat_host, env_repository_eudat_token)
-                            data['deposition_url'] = ''.join((env_repository_eudat_host.replace('api', 'records/'), data['deposition_id']))
-                            eudat_add_zip_to_depot(env_repository_eudat_host, data['deposition_id'], file_name, compendium_files, env_repository_eudat_token)
-                            # Add metadata that are in compendium in db:
-                            if 'metadata' in current_compendium:
-                                if 'eudat' in current_compendium['metadata']:
-                                    md = current_compendium['metadata']['eudat']
-                                    eudat_update_md(env_repository_eudat_host, data['deposition_id'], md, env_repository_eudat_token)
+                    # Aquire path to files via env var and id:
+                    compendium_files = os.path.normpath(os.path.join(env_compendium_files, data['compendium_id']))
+                    # Determine state of that compendium: Is is a bag or not, zipped, valid, etc:
+                    compendium_state = files_scan_path(compendium_files)
+                    if not compendium_state == 0:
+                        # Case path does not exist:
+                        if compendium_state == 1:
+                            # Case: Is a bagit bag:
+                            logging.getLogger('bagit').setLevel(logging.CRITICAL)
+                            try:
+                                bag = bagit.Bag(compendium_files)
+                                bag.validate()
+                                status_note(''.join(('Valid bagit bag at <', str(data['compendium_id']), '>')))
```

Comment:
What do you think about moving the `''.join` logic into `status_note`, so that the function accepts both arrays and simple strings (if that is possible, I'm just guessing) ?

Reply:
I was thinking the same but was too lazy to implement because it is error-prone

Reply:
Fair enough.

Reply:
Good that you mentioned it: while upgrading the status msg I found several lines, where a returned `None` would have crashed the shipper. I'll redo the status lines now completely.